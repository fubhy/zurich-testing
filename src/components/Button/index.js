import React, { Component } from 'react';

const MyChildComponent = ({ children }) => (
  <span>{children}</span>
);

class Button extends Component {
  state = {
    color: this.props.inactive || 'green',
  };

  handleClick = () => {
    this.setState({
      color: this.props.active || 'red',
    });
  }

  render() {
    return (
      <button style={{
        backgroundColor: this.state.color
      }} onClick={this.handleClick}>
        <MyChildComponent>Click me!</MyChildComponent>
      </button>
    );
  }
}

export default Button;
