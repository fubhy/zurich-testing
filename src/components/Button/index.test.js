import React from 'react';
import Button from './index';

describe('Button', () => {
  it('renders with correct text', () => {
    const component = shallow(<Button />);
    const text = 'Click me!';

    expect(component.contains(text)).toEqual(true);
  });  

  // We can use snapshot testing.
  it('renders correctly', () => {
    const component = shallow(<Button />);
    expect(component).toMatchSnapshot();
  });

  // Notice the difference between shallow and mount.
  it('renders correctly with prop', () => {
    const component = mount(<Button inactive="orange" active="blue" />);
    expect(component).toMatchSnapshot();
  });

  // We can use the convenience matchers from jest-enzyme.
  it('initializes the state with the given props', () => {
    const component = shallow(<Button inactive="orange" active="blue" />);
    expect(component).toHaveState('color', 'orange');
  });

  // We can simulate click events on the rendered components.
  it('changes the color on click', () => {
    const component = shallow(<Button />);

    // Color is 'green' before the click.
    expect(component).toHaveStyle('backgroundColor', 'green');
    component.simulate('click');

    // Color is 'red' after the click.
    expect(component).toHaveStyle('backgroundColor', 'red');
  });

  it('changes to a custom color on click', () => {
    const component = shallow(<Button inactive="orange" active="blue" />);
    
    // Color is 'green' before the click.
    expect(component).toHaveStyle('backgroundColor', 'orange');
    component.simulate('click');

    // Color is 'blue' after the click.
    expect(component).toHaveStyle('backgroundColor', 'blue');
  });
});
