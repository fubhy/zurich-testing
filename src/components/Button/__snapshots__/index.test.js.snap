// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders correctly 1`] = `
<button
  onClick={[Function]}
  style={
    Object {
      "backgroundColor": "green",
    }
  }
>
  <MyChildComponent>
    Click me!
  </MyChildComponent>
</button>
`;

exports[`Button renders correctly with prop 1`] = `
<button
  onClick={[Function]}
  style={
    Object {
      "backgroundColor": "orange",
    }
  }
>
  <span>
    Click me!
  </span>
</button>
`;
